### PR TITLE
New Keyboard UI implementation

### LINKED ISSUE #84: Could we please have a shift key on the right in the keyboard?
a shift key on the right enables me to type my passwords more quickly!

### LINKED ISSUE #256: English Keyboard input improvement suggestions for FxR
Hardware: All
Steps to Reproduce: N/A
Actual Behavior: N/A
Expected Behavior:
1. Expected to have "/" key available on main panel for easy URL entry
2. Expected to have ".com" button available to speed entry
3. Expected previously entered URL text to remain (so it is possible to add to it), and not have it erased
Recommendation:
1. / Key on main keyboard
2. .com button on main keyboard
3. URL text remains when selected unless backspace is selected (this presumes text is already selected)

### LINKED ISSUE #333: Add "." period key to the numeric/symbols keyboard.
Hardware:
Oculus Go

Steps to Reproduce:
1. Click on URL bar,
2. Navigate to any IP-based URL and note how you have to go between ABC and 123 keyboards for the period "." key -- which is slow.

Recommendation:
Add period key to numeric/symbols keyboard.  Also would be nice if we just combined numeric/alpha keyboards since we are not short on space.

### PR DESCRIPTION
**Pull request by keianhzo**
Fixes #256, Fixes #84, Fixes #333 Added new Keyboard UI

The keyboard code really needs a refactoring, I'll address that in the next keyboard iteration.

### CODE REVIEW

**Comment by bluemarvin** [comment]
Minor nit, but on daydream, the expanded `h` key menu is clipped on the right side.
clip

**Comment by keianhzo** [comment]
Fixed closing the pop up when clicking on the keyboard padding.

File: app/src/common/shared/org/mozilla/vrbrowser/ui/KeyboardWidget.java [hunk 1]
```diff
@@ -229,50 +261,59 @@ public void onPress(int primaryCode) {
...
-            mOriginalLayoutParams = (RelativeLayout.LayoutParams) mKeyboardview.getLayoutParams();
             RelativeLayout.LayoutParams params = new RelativeLayout.LayoutParams(
                     RelativeLayout.LayoutParams.WRAP_CONTENT,
                     RelativeLayout.LayoutParams.WRAP_CONTENT);
 
+            boolean rightAligned = false;
+            int maxCharsPerLine = MAX_CHARS_PER_LINE_LONG;
             StringBuilder popupCharacters = new StringBuilder(popupKey.popupCharacters);
             switch (popupKey.codes[0]) {
+                case 110: {
```
Comment (bluemarvin) [inline app/src/common/shared/org/mozilla/vrbrowser/ui/KeyboardWidget.java]: part of the refactor should be replacing these numbers with const values so it is easier to read.

**Comment by bluemarvin** [comment]
@MortimerGoro Feel free to review the code at your leisure and file any issues as follow ups.

**Pull request merged**